lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -329,32 +330,28 @@ def run_sync(command, stdin = nil)
       stdout = ''
       stderr = ''
       exit_status = nil
-      started = false
 
-      channel = session.open_channel do |ch|
-        ch.on_data do |c, data|
-          stdout.concat(data)
-        end
-        ch.on_extended_data { |_, _, data| stderr.concat(data) }
-        ch.on_request('exit-status') { |_, data| exit_status = data.read_long }
-        # Send data to stdin if we have some
-        ch.send_data(stdin) unless stdin.nil?
-        # on signal: sending the signal value (such as 'TERM')
-        ch.on_request('exit-signal') do |_, data|
-          exit_status = data.read_string
-          ch.close
-          ch.wait
-        end
-        ch.exec command do |_, success|
-          raise 'could not execute command' unless success
-
-          started = true
+      master, slave = PTY.open
+      read, write = IO.pipe
+      pid = spawn('/usr/bin/ssh', @ssh_user + '@' + @host, command, :in => read, :out => slave)
```

Comment:
The old implementation was setting a couple of options when opening the connection https://github.com/theforeman/smart_proxy_remote_execution_ssh/blob/master/lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb#L271 . We need to do something similar here. Some of them have direct counterpart in ssh's options and we should use those. Take a look at `man 1 ssh`, search for `-o option`

Reply:
I assume that option `ssh_options[:paranoid] = true` is deprecated. But what about this option:
```ruby
ssh_options[:logger] = Proxy::RemoteExecution::Ssh::LogFilter.new(Proxy::Dynflow::Log.instance)
```
How can be this option included here?

Reply:
> I assume that option ssh_options[:paranoid] = true is deprecated. 

It is "deprecated", it just got renamed to something else in later net-ssh versions. I think we want something like `StrictHostKeyChecking=accept-new` would be a fitting equivalent.

> ssh_options[:logger]

We can't really do anything about this one, but it was always more of an workaround so no action is needed here.